### PR TITLE
Happy Thoughts 

### PR DESCRIPTION
**Pull request by moablomkvist**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -7,6 +7,23 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
...
+  hearts: {
+    type: Number,
+    default: 0,
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+});
+
```
Comment (carolinebirgersson) [inline server.js:26]: Good naming, easy to follow!

File: server.js [hunk 2]
```diff
@@ -7,6 +7,23 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
```
Comment (carolinebirgersson) [inline server.js:7]: I'm sure you know, but just a reminder to be consistent with the use of ";", even in the imports.

File: server.js [hunk 3]
```diff
@@ -23,6 +40,42 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+app.get('/thoughts', async (req, res) => {
+  const thoughts = await Thought.find().sort({createdAt: 'desc'}).limit(20).exec();
+  res.json(thoughts);
+});
+
+app.post('/thoughts', async (req, res) => {
+  const {message, hearts} = req.body;
```
Comment (carolinebirgersson) [inline server.js:49]: You use spaces after the {} and the naming further down (ex. { thoughtID }), try to be consistent all the way through ☺️

File: server.js [hunk 4]
```diff
@@ -23,6 +40,42 @@ app.get('/', (req, res) => {
...
+  const thought = new Thought({message: message, hearts: hearts});
+
+  try {
+    const savedThought = await thought.save();
+    res.status(201).json(savedThought);
+  } catch (err) {
+    res.status(400).json({ message: 'Could not post your happy thought', 
+    error: err.errors
+  });
```
Comment (carolinebirgersson) [inline server.js:58]: Suitable error message, good choice of words!

File: server.js [hunk 5]
```diff
@@ -23,6 +40,42 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+app.get('/thoughts', async (req, res) => {
+  const thoughts = await Thought.find().sort({createdAt: 'desc'}).limit(20).exec();
+  res.json(thoughts);
+});
```
Comment (carolinebirgersson) [inline server.js:46]: Great use of methods here with both find, sort, limit and exec! 👏🏼

File: server.js [hunk 6]
```diff
@@ -23,6 +40,42 @@ app.get('/', (req, res) => {
...
+    res.status(201).json({ message: 'Could not find thought', 
+    error: err.errors
+  });
+  } catch (err) {
+    res.status(404).json({ message: 'Could not find a thought with that ID', 
+    error: err.errors,
+    });
+  }
+});
+
```
Comment (carolinebirgersson) [inline server.js:78]: Nice use of both 201 and 404, with clear errors!

**Review by carolinebirgersson — COMMENTED** [review]
Clean and tidy code that's easy to follow! 🌟 You covered all the things needed and did it great!

**Review by KarolCislowski — APPROVED** [review]
Very nice and clean app 👍🏻

**Pull request closed without merging**